Assets/_BForBoss/Scripts/Debug/DebugWindow.cs

```diff
@@ -0,0 +1,172 @@
+            }
+            else
+            {
+                _instance = this;
+            }
+            
+            DontDestroyOnLoad(this);
+            
+            foreach (Type viewType in Assembly.GetAssembly(typeof(DebugView)).GetTypes())
+            {
+                if (viewType.IsClass && !viewType.IsAbstract && viewType.IsSubclassOf(typeof(DebugView)))
+                {
+                    _debugOptions.Add(viewType);
+                }
+            }
+        }
+
+        private void Update()
+        {
+            if (Keyboard.current[_keyCodeModifier].isPressed &&
+                Keyboard.current[_keyCodeCharacter].wasPressedThisFrame)
+            {
+                if (_isPanelShowing)
+                {
+                    ClosePanel();
+                }
+                else
+                {
+                    OpenPanel();
+                }
+
+                _isPanelShowing = !_isPanelShowing;
+            }
+        }
+
+        private void OnGUI()
+        {
+            if (!_isPanelShowing)
+            {
+                return;
+            }
+
+            if (_currentDebugView != null && _currentDebugView.IsInitialized)
+            {
+                _currentDebugView.DrawGUI();
+            }
+            else
+            {
+                CreateDebugViewList();
+            }
+        }
+
+        private void CreateDebugViewList()
+        {
+            using (new GUILayout.AreaScope(_windowRect, "Debug Options", GUI.skin.window))
+            {
+                using (new GUILayout.HorizontalScope())
+                {
+                    GUILayout.FlexibleSpace();
+
+                    using (new GUILayout.VerticalScope())
+                    {
+                        GUILayout.Space(0.15f * _windowRect.height);
+
+                        foreach (Type debugType in _debugOptions)
+                        {
+                            if (GUILayout.Button(debugType.Name))
+                            {
+                                ConstructorInfo constructorInfo = debugType.GetConstructor(new Type[] {typeof(Rect)});
+                                object[] parameters = new object[] {_windowRect};
+                                _currentDebugView = constructorInfo.Invoke(parameters) as DebugView;
+                            }
+                        }
+                    }
+                    
+                    GUILayout.FlexibleSpace();
+                }
+                
+                GUILayout.FlexibleSpace();
+            }
+        }
+
+        private void ResetView()
+        {
+            if (_currentDebugView != null)
+            {
+                _currentDebugView.ResetData();
+                _currentDebugView = null;
+            }
+        }
+
+        private void OpenPanel()
+        {
+            _currentState = _stateManager.GetState();
+            _stateManager.SetState(State.Pause);
+            FirstPersonPlayer.IsDebugWindowOpen = true;
```

Comment:
```suggestion
            IInputSettings input = FindObjectOfType<FirstPersonPlayer>();
            input.DisableActions();
```

Disabling/Enabling the Action within IInputSettings might be more cleaner, since we'll be needing it in Other settings menu

Reply:
Inside IInputSettings
```suggestion
                   public void DisableActions()
        {
            controllerLookInputAction.Disable();
            mouseLookInputAction.Disable();
        }

        public void EnableActions()
        {
            controllerLookInputAction.Enable();
            mouseLookInputAction.Enable();
        }
```

Reply:
Oh good point, made a similar script as Im finishing the Pause menu haha. Should just use this instead haha